### PR TITLE
Add ability to instantiate tokenizers from the Hub (from_pretrained)

### PR DESCRIPTION
**Pull request by n1t0**
Similarly to what exists in `transformers`, add the ability to instantiate tokenizers from `tokenizer.json` files uploaded to the Hugging Face Hub.

Example:
```python
from tokenizers import Tokenizer
tokenizer = Tokenizer.from_pretrained('bert-base-uncased')
```

### CODE REVIEW

File: tokenizers/src/utils/from_pretrained.rs [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    pub revision: String,
+    pub user_agent: HashMap<String, String>,
+    pub auth_token: Option<String>,
+}
+
+impl Default for FromPretrainedParameters {
+    fn default() -> Self {
+        Self {
+            revision: "main".into(),
+            user_agent: HashMap::new(),
```
Comment (Narsil) [inline tokenizers/src/utils/from_pretrained.rs:93]: Shouldn't this be "tokenizers", "rust" or something by default ?

**Review by Narsil — COMMENTED** [review]
Shouldn't we add 1 test per binding + 1 for rust at least ?
It will depend on network+hub which is never great but I think it could be helpful.

Also we probably need to add it to the docs, no ?

File: tokenizers/src/utils/from_pretrained.rs (hunk 1 above)
Comment (n1t0) [reply tokenizers/src/utils/from_pretrained.rs:93]: It is, by default the user agent always contains `tokenizers/{RUST_VERSION}`, this is for additional items

**Comment by n1t0** [comment]
> Shouldn't we add 1 test per binding + 1 for rust at least ?
It will depend on network+hub which is never great but I think it could be helpful.

Agreed

> Also we probably need to add it to the docs, no ?

I've already added it in every place I though about. Do you have something specific in mind?

**Comment by Narsil** [comment]
> I've already added it in every place I though about. Do you have something specific in mind?

I missed it in `quicktour` that was what I had in mind.

**Pull request merged**